### ISSUE TITLE
bug: $ionicScrollDelegate cause premature keyboard close

### ISSUE THREAD

**Issue by jfspencer** [driftyco/ionic#3149]
**Type**: <span ionic-type>bug</span>

**Platform**: <span ionic-platform>ios</span> <span ionic-platform-version>8</span>  <span ionic-webview>webview</span>

<span ionic-description>using scrollTo and scrollTop with $ionicScrollDelegate, and then focusing on a text input causes the native keyboard to open then immediately close. This may be related to #2902 digest cycle, or it could be caused by the very long animation decay that is occurring. Since reproduction is limited to emulators/devices I have linked a video showing the issue and the applicable code is below

video reproducing the issue
https://drive.google.com/file/d/0B0-I41uSyatNMXRja3dxa2tpUm8/view?usp=sharing

    <ion-content direction="y" on-scroll="scrollChange()" scroll-event-interval="100" start-y="56">...

    //function being called by on-scroll
    $scope.scrollChange = function(){
            var position = $ionicScrollDelegate.$getByHandle("boardListScroll").getScrollPosition();
            if(position.top < 40)
                $ionicScrollDelegate.$getByHandle("boardListScroll").scrollTop(true);
            if(position.top >= 40 && position.top < 70) {
                $ionicScrollDelegate.$getByHandle("boardListScroll").scrollTo(null, 56, true);
            }
    };

<span is-issue-template></span>

**Comment by jfspencer** [comment]
using on-scroll-complete solves the issue.

**Comment by guilhermegm** [comment]
@jfspencer 

Hello, I had the same issue and I don't understand where to use this "on-scroll-complete", could you give more detail ? Thanks

**Comment by jfspencer** [comment]
I was implementing a snapping search bar so I only needed to call
on-scroll-complete at the end of the scroll to snap open or closed. If you
need to have a function call during the scroll action and involve an input
then it would still be broken. I'll reopen the issue for that case.

On Saturday, February 21, 2015, guilhermegm <notifications@github.com>
wrote:

> @jfspencer <https://github.com/jfspencer>
>
> Hello, I had the same issue and I don't understand where to use this
> "on-scroll-complete", could you give more detail ? Thanks
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/driftyco/ionic/issues/3149#issuecomment-75378163>.
>


-- 

Cheers,


James

**Comment by jfspencer** [comment]
If a function call needs to be triggered on-scroll while interacting with an input, still causes the keyboard to open then immediately close.

**Comment by insraq** [comment]
I believe this is related to Issue #2745 and this commit should have fixed it: 0145dc3.

Basically in beta.14, all `scrollTo` methods will blur the input element. It is not released yet I believe. As a workaround, you can do this manually

```javascript
$ionicScrollDelegate.$getByHandle('boardListScroll').getScrollView().scrollTo(0, 0, true);
```

**Comment by Ionitron** [comment]
Greetings @jfspencer!

I'm sorry, but my sensors indicate that this issue may no longer apply to recent versions of Ionic.

If you are still experiencing this issue, please feel free to reopen it by creating a new issue and provide the following information, so that I may assist you:

- Can you provide a detailed description of this issue?
- How would a robot go about replicating this issue?
- On which platform does this occur? (iOS, Android, all, etc.)
- On which platform versions does this occur? (Android 4.2, iOS 7, all, etc.)
- Is this an issue with the Ionic Framework or Ionic CLI?

Additionally, providing a [codepen](http://codepen.io/ionic/public-list/) that replicates the issue is extremely helpful.

If you wish to get help using the framework itself, I recommended visiting [the forum](http://forum.ionicframework.com). There are many humans there who can offer sage advice.

Thank you for allowing me to assist you.
